R/tbl_split.R

```diff
@@ -1,36 +1,125 @@
+#'
+#' Footnotes and caption handling are experimental and may change in the future.
+#'
+#' @examplesIf identical(Sys.getenv("NOT_CRAN"), "true") || identical(Sys.getenv("IN_PKGDOWN"), "true")
+#' # Example 1 ----------------------------------
+#' # Split by rows
+#' trial |>
+#'   tbl_summary(by = trt) |>
+#'   tbl_split_by_rows(variables = c(marker, grade)) |>
+#'   tail(n = 1) # Print only last table for simplicity
+#'
+#' # Example 2 ----------------------------------
+#' # Split by rows with row numbers
+#' trial |>
+#'   tbl_summary(by = trt) |>
+#'   tbl_split_by_rows(row_numbers = c(5, 7)) |>
+#'   tail(n = 1) # Print only last table for simplicity
+#'
+#' # Example 3 ----------------------------------
+#' # Split by columns
+#' trial |>
+#'   tbl_summary(by = trt, include = c(death, ttdeath)) |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2")) |>
+#'   tail(n = 1) # Print only last table for simplicity
+#'
+#' # Example 4 ----------------------------------
+#' # Both row and column splitting
+#' trial |>
+#'   tbl_summary(by = trt) |>
+#'   tbl_split_by_rows(variables = c(marker, grade)) |>
+#'   tbl_split_by_columns(groups = list("stat_1", "stat_2")) |>
+#'   tail(n = 1) # Print only last table for simplicity
+#'
+#' # Example 5 ------------------------------
+#' # Split by rows with footnotes and caption
+#' trial |>
+#'   tbl_summary(by = trt, missing = "no") |>
+#'   modify_footnote_header(
+#'     footnote = "All but four subjects received both treatments in a crossover design",
+#'     columns = all_stat_cols(),
+#'     replace = FALSE
+#'   ) |>
+#'   modify_footnote_body(
+#'     footnote = "Tumor grade was assessed _before_ treatment began",
+#'     columns = "label",
+#'     rows = variable == "grade" & row_type == "label"
+#'   ) |>
+#'   modify_spanning_header(
+#'     c(stat_1, stat_2) ~ "**TRT**"
+#'   ) |>
+#'   modify_abbreviation("I = 1, II = 2, III = 3") |>
+#'   modify_caption("_Some caption_") |>
+#'   modify_footnote_spanning_header(
+#'     footnote = "Treatment",
+#'     columns = c(stat_1)
+#'   ) |>
+#'   modify_source_note("Some source note!") |>
+#'   tbl_split_by_rows(variables = c(marker, stage, grade), footnotes = "last", caption = "first") |>
+#'   tail(n = 2) |>
+#'   head(n = 1) # Print only one but not last table for simplicity
+#'
+#' @name tbl_split_by
 NULL
 
 #' @export
-#' @rdname tbl_split
-tbl_split <- function(x, ...) {
-  UseMethod("tbl_split")
-}
-
-#' @export
-#' @rdname tbl_split
-tbl_split.gtsummary <- function(x, variables, ...) {
+#' @rdname tbl_split_by
+tbl_split_by_rows <- function(x, variables,
+                              row_numbers,
+                              footnotes = c("all", "first", "last"),
+                              caption = c("all", "first")) {
   set_cli_abort_call()
-  check_dots_empty()
+
+  # list map -------------------------------------------------------------------
+  if (inherits(x, "list")) {
```

Comment:
Shouldn't we be checking for class `tbl_split` here? is there an advantage to keeping it more generic?

Reply:
I was thinking that some people may want to use their custom lists of different tables